crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1022,6 +1023,16 @@ where
 
                 // we can update the FCU blocks
                 let _ = self.update_canon_chain(&target);
+                return
+            }
+
+            // if the inserted block is the currently targeted `finalized` or `safe` block, we will
+            // attempt to make them canonical, because they are also part of the canonical chain and
+            // their missing block range might already be downloaded (buffered).
+            if let Some(target_hash) =
+                ForkchoiceStateHash::find(&target, inserted.hash).filter(|h| !h.is_head())
+            {
+                let _ = self.blockchain.make_canonical(target_hash.as_ref());
```

Comment:
this will only run if the previous `make_canonical` call returns `Err` right? Should we try to make the safe / finalized blocks canonical if this fails for a reason other than "head block not found"?